### PR TITLE
fix(jsonrpc): add blockNumOrHash length check 

### PR DESCRIPTION
**Pull request by Sunny6889**
**What does this PR do?**
* **Bug Fixes**: Added input validation for JSON-RPC block identifiers and hex parsing to reject null/overlong inputs, reducing risk of malformed-request abuse.
* **Chores**: Centralized rate limiter default values and lowered P2P rate limiter defaults (global and per-IP) for safer defaults.

**Why are these changes required?**
- For safety

**This PR has been tested by:**
- Unit Tests: Expanded unit and JSON-RPC tests covering validation, parsing, edge cases, and overflow scenarios.
- Manual Testing

**Follow up**

**Extra details**

## Summary by CodeRabbit

* **Bug Fixes**
  * Added validation to reject null or overly long block identifiers and JSON-hex inputs; improved DDoS/malformed-request resilience.
  * Improved cross-thread visibility for an in-memory DB cursor to reduce concurrency issues.

* **Chores**
  * Centralized rate-limiter defaults and lowered P2P/global per-IP defaults for safer runtime defaults.

* **Tests**
  * Expanded unit and JSON-RPC tests covering validation, parsing, edge cases, and overflow scenarios.

### CODE REVIEW

File: framework/src/main/java/org/tron/core/services/jsonrpc/TronJsonRpcImpl.java [hunk 1]
```diff
@@ -393,6 +396,9 @@ public String getLatestBlockNum() {
   @Override
   public String getTrxBalance(String address, String blockNumOrTag)
       throws JsonRpcInvalidParamsException {
+    // Add length check and validate hex format to prevent DDoS attacks
```
Comment (0xbigapple) [inline framework/src/main/java/org/tron/core/services/jsonrpc/TronJsonRpcImpl.java:400]: Replacing the hex parsing validation with a pure length check is not equivalent.

File: framework/src/main/java/org/tron/core/services/jsonrpc/TronJsonRpcImpl.java
Comment (0xbigapple) [inline framework/src/main/java/org/tron/core/services/jsonrpc/TronJsonRpcImpl.java:783]: lost `validateBlockNumOrHashOrTag` ？

File: framework/src/main/java/org/tron/core/services/jsonrpc/TronJsonRpcImpl.java [hunk 2]
```diff
@@ -589,12 +583,6 @@ public String getABIOfSmartContract(String contractAddress, String blockNumOrTag
       }
 
     } else {
-      try {
```
Comment (0xbigapple) [inline framework/src/main/java/org/tron/core/services/jsonrpc/TronJsonRpcImpl.java:592]: why remove this check?

File: framework/src/main/java/org/tron/core/services/jsonrpc/JsonRpcApiUtil.java [hunk 3]
```diff
@@ -85,6 +90,13 @@ public static String getMethodSign(String method) {
     return Hex.toHexString(selector);
   }
 
+  public static void validateBlockNumOrHashOrTag(String input)
+      throws JsonRpcInvalidParamsException {
+    if (input == null || input.length() > MAX_BLOCK_IDENTIFIER_LENGTH) {
+      throw new JsonRpcInvalidParamsException("input is null or length exceeds maximum allowed");
```
Comment (0xbigapple) [inline framework/src/main/java/org/tron/core/services/jsonrpc/JsonRpcApiUtil.java:96]: what about also check tags and BlockNumer in this function?

File: framework/src/main/java/org/tron/core/services/jsonrpc/TronJsonRpcImpl.java
Comment (0xbigapple) [inline framework/src/main/java/org/tron/core/services/jsonrpc/TronJsonRpcImpl.java:571]: lost `validateBlockNumOrHashOrTag` check

Comment (0xbigapple) [inline framework/src/main/java/org/tron/core/services/jsonrpc/TronJsonRpcImpl.java:537]: lost `validateBlockNumOrHashOrTag` check？

Comment (0xbigapple) [inline framework/src/main/java/org/tron/core/services/jsonrpc/TronJsonRpcImpl.java:298]: lost blockHash.length check?

File: framework/src/main/java/org/tron/core/services/jsonrpc/JsonRpcApiUtil.java (hunk 3 above)
Comment (Sunny6889) [reply framework/src/main/java/org/tron/core/services/jsonrpc/JsonRpcApiUtil.java:96]: Hi good idea, but because want to limit the PR changes to only one scope a time

File: framework/src/main/java/org/tron/core/services/jsonrpc/TronJsonRpcImpl.java (hunk 2 above)
Comment (Sunny6889) [reply framework/src/main/java/org/tron/core/services/jsonrpc/TronJsonRpcImpl.java:592]: As I think this check is no need and cost computational resources.
